python/lsst/sims/featureScheduler/mockTelem/mock_observatory.py

```diff
@@ -0,0 +1,485 @@
+                                                       observation['sunRA'], observation['sunDec'])
+        observation['moonPhase'] = sun_moon_info['moon_phase']
+
+        observation['ID'] = self.obsID_counter
+        self.obsID_counter += 1
+
+        return observation
+
+    def check_mjd(self, mjd, cloud_skip=20.):
+        """See if an mjd is ok to observe
+        Parameters
+        ----------
+        cloud_skip : float (20)
+            How much time to skip ahead if it's cloudy (minutes)
+
+
+        Returns
+        -------
+        bool
+
+        mdj : float
+            If True, the input mjd. If false, a good mjd to skip forward to.
+        """
+        passed = True
+        new_mjd = mjd + 0
+        delta_t = (mjd-self.mjd_start)*24.*3600.
+        clouds = self.cloud_model.get_cloud(delta_t+0.)
+        # Maybe set this to a while loop to make sure we don't land on another cloudy time?
+        # or just make this an entire recursive call?
+        if clouds > self.cloud_limit:
+            passed = False
+            new_mjd = mjd + cloud_skip/60./24.
+        alm_indx = np.searchsorted(self.almanac.sunsets['sunset'], mjd) - 1
+        # at the end of the night, advance to the next setting twilight
+        if mjd > self.almanac.sunsets['sun_n12_rising'][alm_indx]:
+            passed = False
+            new_mjd = self.almanac.sunsets['sun_n12_setting'][alm_indx+1]
+        if mjd < self.almanac.sunsets['sun_n12_setting'][alm_indx]:
+            passed = False
+            new_mjd = self.almanac.sunsets['sun_n12_setting'][alm_indx+1]
+        # We're in a down night, advance to next night
+        if self.almanac.sunsets['night'][alm_indx] in self.down_nights:
+            passed = False
+            new_mjd = self.almanac.sunsets['sun_n12_setting'][alm_indx+1]
+        # recursive call to make sure we skip far enough ahead
+        if not passed:
+            while not passed:
+                passed, new_mjd = self.check_mjd(new_mjd)
+            return False, new_mjd
+        else:
+            return True, mjd
+
+    def observe(self, observation):
+        """Try to make an observation
+
+        Returns
+        -------
+        status : bool
+            Result of if the observation worked
+        observation : observation object
+            None if there was no observation taken. Completed observation with meta data filled in.
+        new_night : bool
+            Have we started a new night.
+        """
+
+        start_night = self.night.copy()
+
+        # Make sure the kinematic model is set to the correct mjd
+        t = Time(self.mjd, format='mjd')
+        self.observatory.update_state(t.unix)
+
+        # XXX--temparary fix to keep the camera at rotation angle zero.
+        # XXX--TODO:  Move rotation logic to the scheeduler
+        position = self.observatory.radecang2position(self.observatory.dateprofile, observation['RA'],
```

Comment:
Should we be passing the rotator angle here, even if we are not setting it yet? I think it would be better to hard code that zero in the place we'll work on later, so we don't forget it afterwards...

Reply:
yeah, this is the very next thing to change. Adding in the logic for setting the rotation angle is big enough (I think we want a new class) I just wanted to get this branch merged before tackling it.